corehq/apps/data_interfaces/models.py

```diff
@@ -503,7 +503,8 @@ def _get_case_ids_from_postgres(cls, domain, case_type, boundary_date=None, db=N
             for c_id in CommCareCaseSQL.objects.using(db).filter(q_expression).values_list('case_id', flat=True):
                 yield c_id
         else:
-            for c_id in run_query_across_partitioned_databases(CommCareCaseSQL, q_expression, values=['case_id']):
+            for pk, c_id in paginate_query_across_partitioned_databases(
+                    CommCareCaseSQL, q_expression, values=['case_id']):
```

Comment:
What do you think of having `paginate_query_across_partitioned_databases` raise an error if `values=[...]` is specified, but `pk` is not one of the fields? That way it would force explicitly requesting `pk` or fail immediately.

Reply:
I removed the `pk` from the return value. It was not difficult to remove. I think I kept it in there the first time to avoid the list splice. https://github.com/dimagi/commcare-hq/pull/24717/commits/a75561a18e40d09fa9560663c1e723c406051e49